src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -52,6 +60,7 @@ public sealed partial class JsonSerializerOptions
         public JsonSerializerOptions()
         {
             Converters = new ConverterList(this);
+            TrackOptionsInstance(this);
```

Comment:
Since this is only used in development scenarios, not production (my understanding) can we detect this mode and only track in development scenarios?

Reply:
That's what the TODO higher up is about. I agree that's desirable.